### ISSUE TITLE
Nginx installation documentation doesnt work.

### ISSUE THREAD

**Issue by dlannan-fmad** [openresty/lua-nginx-module#1829]
github checkout of lua-nginx-modile
nginix version 1.19.0
opentry luajit2-2.1-20201027

I followed the instructions to install with nginx and I get this:
```
nginx: [alert] detected a LuaJIT version which is not OpenResty's; many optimizations will be disabled and performance will be compromised (see https://github.com/openresty/luajit2 for OpenResty's LuaJIT or, even better, consider using the OpenResty releases from https://openresty.org/en/download.html)
nginx: [alert] failed to load the 'resty.core' module (https://github.com/openresty/lua-resty-core); ensure you are using an OpenResty release from https://openresty.org/en/download.html (reason: module 'resty.core' not found:
        no field package.preload['resty.core']
        no file './resty/core.lua'
        no file '/usr/local/share/luajit-2.1.0-beta3/resty/core.lua'
        no file '/usr/local/share/lua/5.1/resty/core.lua'
        no file '/usr/local/share/lua/5.1/resty/core/init.lua'
        no file './resty/core.so'
        no file '/usr/local/lib/lua/5.1/resty/core.so'
        no file '/usr/local/lib/lua/5.1/loadall.so'
        no file './resty.so'
        no file '/usr/local/lib/lua/5.1/resty.so'
        no file '/usr/local/lib/lua/5.1/loadall.so') in /opt/fmadio/www/conf/nginx_http.conf:142
``
The error states to use the openresty luajit - but it is. I followed the instructions to do so. And there is no other luajit installed on the system. I have tried various different tweaks, but I cannot seem to get this to work. 
It looks like rest.core is _required_ to use this module in nginx. Is that the case? 
I do not wish to add openresty (its far too big for our implementation). If openresty is now required I'll add my own luajit module. 

Any help would be greatly appreciated.

**Comment by dlannan-fmad** [comment]
I think it must be related to our platform. We are using tiny core linux, with many customizations. 
Going to check environment variables and paths. 
Im still not quite understanding why the package search path is looking in '/usr/local/share/luajit-2.1.0-beta3/resty/core.lua' when the recommended luajit is 2-2.1 ? Could this be an issue?

**Comment by dlannan-fmad** [comment]
After many attempts, it doesnt appear that the installation procedure for nginx works.
There is alwas a dependency to load resty.core when exectued. 
![image](https://user-images.githubusercontent.com/60128766/100518146-bb03f980-31df-11eb-9d04-5747f7ce6ec2.png)
We do not want to use openresty - can someone confirm if this module works as a standalone nginx module or not?

**Comment by dlannan-fmad** [comment]
Ok. More investigation into this build process, and it looks like there is a dependency on the lua resty core repo:
https://github.com/openresty/lua-resty-core
Which means the install documents will not allow building of the module with the installation of the above repo. 

Quick question: Is there a bundle that can be used with the nginx module? Rather than a bunch of lua script spread across various directories?

**Comment by dlannan-fmad** [comment]
I have found a solution, and it seems to work (obviously resty.core features will not work). 
Comment out lines 3996-4002 here:
https://github.com/openresty/lua-nginx-module/blob/7105adaa523adedec80f0aaa13388b88d08988f8/src/ngx_http_lua_util.c#L3996
And comment out:
https://github.com/openresty/lua-nginx-module/blob/7105adaa523adedec80f0aaa13388b88d08988f8/src/ngx_http_lua_util.c#L3924

This will allow you to build this module without the need of installing the full openresty system.

**Comment by Firegarden** [comment]
Amazing how many people have this problem. I believe the solution is simple - you need update your nginx.conf to include the lua package path as follows

`lua_package_path "/usr/local/lib/lua/?.lua;;";`

**Comment by dlannan** [comment]
> Amazing how many people have this problem. I believe the solution is simple - you need update your nginx.conf to include the lua package path as follows
> 
> `lua_package_path "/usr/local/lib/lua/?.lua;;";`

This wont help with building _without_ resty.core. Which is what is mentioned in the comment - no openresty, just the lua module.

**Comment by Firegarden** [comment]
@dlannan sorry I did not read the thread. Thank you for your correction. 

I am a little confused if you wish to avoid _OPENRESTY_ or avoid _RESTY CORE_ the reason I ask is because I don't use openresty rather I just download resty-core eg. `sudo wget https://github.com/openresty/lua-resty-core/archive/v${LUA_RESTY_CORE}.tar.gz`

this is significantly less than having to build the full `openresty` 

note I did find I need to now download lua-resty-lrucache as well to get it all working.

so in summary in my build I use the following and it works fine

- LUA_JIT_VER=2.1-20201229 
- LUA_NGINX_VER=0.10.17
- LUA_RESTY_CORE=0.1.19
- LUA_LRUCACHE=0.10

**Comment by dlannan** [comment]
No probs. Our use case is a little erm.. specific :) There are a couple of modules, and I have developed some addon modules for doing quite bespoke interactions with lua all embedded into static libs (hence the need to hand build most of this). We dont run any shared libs.
